### ISSUE TITLE
[BUG] Create new item with item_crafted_id instead of item id on SetMetadata

### ISSUE THREAD

**Issue by ZambrOvosky** [VORPCORE/vorp_inventory-lua#209]
When you have more than 1 item stacked, let's say 2 pickaxes, and wants to set the metadata for 1 of them, the `item_id` on `items_crafted` table is inserted as the `id` on `items_crafted` of the original pickaxe instead of the `id` of the item in `items` table.

**Comment by outsider31000** [comment]
if you have an item with a stack that means the item is same , so  adding metadata to a specific item wouldnt make sense as they are **the same** only unique items wont stack.

**Comment by ZambrOvosky** [comment]
> if you have an item with a stack that means the item is same , so adding metadata to a specific item wouldnt make sense as they are **the same** only unique items wont stack.

That's not true, in my usecase I have "pickaxe" that you can stack up to 5. When I use it, I'm not using the 5 of them, just 1. The metadata holds what's the durability for this specific used item.
Like, if it's not supposed to set metadata for N amount of a stack, why does VORP have a export for it them?

**Comment by outsider31000** [comment]
if you have 5 items stacked they arent unique , meaning they dont have metadata or they all have the same metadata. hence you dont need to get a specifc item to apply, **they are the same** it doesnt matter which item you apply

**Comment by outsider31000** [comment]
they matter when items are unique in which you have exports to get specific items with their unique ids

**Comment by outsider31000** [comment]
![image](https://github.com/VORPCORE/vorp_inventory-lua/assets/87246847/a91dee31-686d-4122-89a6-694481b52529)
 it accepts ammount to set metadata for stacks, you cant indididually set different metadata for a stack using one single call.

**Comment by ZambrOvosky** [comment]
> ![image](https://user-images.githubusercontent.com/87246847/281753624-a91dee31-686d-4122-89a6-694481b52529.png) it accepts ammount to set metadata for stacks, you cant indididually set different metadata for a stack using one single call.

Yes you can
https://github.com/VORPCORE/vorp_inventory-lua/blob/c9174d1631922bef9c54d4d472fd972c1f6c68ef/server/services/inventoryApiService.lua#L678C2-L705C5

The call:
`VorpInv.setItemMetadata(source, itemId, metadata, amount)`
or
`exports.vorp_inventory:setItemMetadata(source, itemId, metadata, amount)`

**Comment by outsider31000** [comment]
i still dont understand whats the issue you facing ? the metdata isnt being applied if the item is in a stack?

**Comment by outsider31000** [comment]
![image](https://github.com/VORPCORE/vorp_inventory-lua/assets/87246847/5494cee5-e177-4c42-a059-ac4b93783ef6)
stacked 
![image](https://github.com/VORPCORE/vorp_inventory-lua/assets/87246847/4ea6b8f7-c94e-41b1-8f49-2d6114b79fd8)
applied metadata on a stack of items

**Comment by ZambrOvosky** [comment]
> When you have more than 1 item stacked, let's say 2 pickaxes, and wants to set the metadata for 1 of them, the `item_id` on `items_crafted` table is inserted as the `id` on `items_crafted` of the original pickaxe instead of the `id` of the item in `items` table.

I explained it on the main thread.

**Comment by ZambrOvosky** [comment]
> ![image](https://user-images.githubusercontent.com/87246847/281760217-5494cee5-e177-4c42-a059-ac4b93783ef6.png) stacked ![image](https://user-images.githubusercontent.com/87246847/281760331-4ea6b8f7-c94e-41b1-8f49-2d6114b79fd8.png) applied metadata on a stack of items

log out and log in, the item is gone

**Comment by outsider31000** [comment]
let me try

**Comment by outsider31000** [comment]
this is why i asked for steps of reproduction, i dont need an explanation, i need steps on how to achieve the issue knowhere you meantioned you loose the item on relog.

**Comment by outsider31000** [comment]
step 1 apply meta to a stack of items
step to 2 relog item is gone.

thats what i wanted. 
and I just tried and indeed happens

**Comment by ZambrOvosky** [comment]
> this is why i asked for steps of reproduction, i dont need an explanation, i need steps on how to achieve the issue knowhere you mentioned you loose the item on relog.

> step 1 apply meta to a stack of items step to 2 relog item is gone.
> 
> thats what i wanted. and I just tried and indeed happens

You got a point, I promise to get better on that.

**Comment by outsider31000** [comment]
thanks.
